independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -119,6 +127,30 @@ static BeanInfo createClassBean(ClassInfo beanClass, BeanDeployment beanDeployme
         return bean;
     }
 
+    private static ScopeInfo inheritScope(ClassInfo beanClass, BeanDeployment beanDeployment) {
+        DotName superClassName = beanClass.superName();
+        while (!superClassName.equals(DotNames.OBJECT)) {
+            ClassInfo classFromIndex = beanDeployment.getIndex().getClassByName(superClassName);
+            if (classFromIndex == null) {
+                // class not in index
+                LOGGER.logf(Logger.Level.WARN, "Arc is not able to determine scope for bean " + beanClass
```

Comment:
Why don't you use `Logger.warnf()`? Also the `f` means you can use a formatted log message - you can use params exactly as in `String.format()`. Finally, I'd try to avoid using `Arc` in log messages.

Reply:
Hm, just because I never considered using `%s` more convenient :)
I'll change it.